client/state/jetpack-connect/actions.js

```diff
@@ -316,6 +321,37 @@ export default {
 			);
 		};
 	},
+	isUserConnected( siteId, siteIsOnSitesList ) {
+		let accessibleSite;
+		return ( dispatch ) => {
+			dispatch( {
+				type: SITE_REQUEST,
+				siteId
+			} );
+			return wpcom.site( siteId ).get()
+			.then( ( site ) => {
+				accessibleSite = site;
+				return wpcom.undocumented().isUserConnected( siteId );
+			} )
+			.then( () => {
+				dispatch( receiveSite( omit( accessibleSite, '_headers' ) ) );
+				dispatch( {
+					type: SITE_REQUEST_SUCCESS,
+					siteId
+				} );
+			} )
+			.catch( ( error ) => {
+				dispatch( {
+					type: SITE_REQUEST_FAILURE,
+					siteId,
+					error
+				} );
```

Comment:
It would be nice to have dedicated action creator functions for `SITE_REQUEST_SUCCESS` and `SITE_REQUEST_FAILURE`, but that doesn't need to happen in this PR.